runtime/packages/modules/balances/assembly/balances-module.ts

```diff
@@ -36,8 +39,47 @@ export class BalancesModule {
         Storage.set(accountId.getAddress(), currentAccountData.toU8a());
 
         return currentAccountData;
-    } 
+    }
 
+    /**
+     * Transfer the given amount from sender to receiver
+     * Note: this is just draft implementation, without necessary checks
+     * @param sender 
+     * @param receiver 
+     * @param amount 
+     */
+    static transfer(sender: AccountId, receiver: AccountId, amount: u64): void {
+        const senderAccData = this.getAccountData(sender);
+        const receiverAccData = this.getAccountData(receiver);
+        const senderNewBalance: UInt128 = new UInt128(u128.sub(senderAccData.getFree().value, u128.fromU64(amount)));
+        const receiverNewBalance: UInt128 = new UInt128(u128.add(receiverAccData.getFree().value, u128.fromU64(amount)));
+        this.setBalance(sender, senderNewBalance, senderAccData.getReserved());
+        this.setBalance(receiver, receiverNewBalance, receiverAccData.getReserved());
+        System.incAccountNonce(sender);
+        Log.info("done transfering: " + amount.toString());
+    }
 
+    /**
+     * Apply extrinsic for the module
+     * @param extrinsic 
+     */
+    static applyExtrinsic(extrinsic: Extrinsic): void{
+        const sender: AccountId = AccountId.fromU8Array(extrinsic.from.toU8a()).result;
+        const receiver: AccountId = AccountId.fromU8Array(extrinsic.to.toU8a()).result;
+        this.transfer(sender, receiver, extrinsic.amount.value);
```

Comment:
I think that it would be a good idea to call `this.validateTransaction` before the `this.tranfer` function

Reply:
I think that you should return error response codes on valid = false